fables/tree.py

```diff
@@ -351,7 +351,7 @@ def mimetype_from_stream(stream: Optional[IO[bytes]]) -> Optional[str]:
     if stream is None:
         return None
 
-    mimebytes = stream.read()
+    mimebytes = stream.read(30000)
```

Comment:
Testing showed that correct mimetypes could be determined at _around_ 30kb. This is a small enough buffer that we shouldn't see any crashes due to excessively large buffers.